stats/perf-stat

```diff
@@ -219,6 +193,64 @@ def parse
   # skip the last record, because the interval hasn't run out
 end
 
+def parse_per_socket_mode(fields)
+  # S0,12,1202404,,mem_load_retired.l1_miss,8004582139,66.65,,
+  socket = nil
+
+  if fields[0][0] == 'S'
+    socket = fields[0]
+    fields.delete_at 1
+    fields.delete_at 0
+  end
+
+  socket
+end
+
+def parse_unit(fields)
+  # 7928832,Bytes,llc_misses.mem_read,8005538025,100.00,,
+  unit = nil
+
+  unit = fields[1] if fields[1] == 'Bytes'
+
+  unit
+end
+
+def parse_metrics(metric_val, fields, socket)
+  # 1.000679357,S0,12,170516599,,l1d_pend_miss.pending,8004582139,66.65,80.5,Load_Miss_Real_Latency
+  metric_key = nil
+
+  if !fields[5].nil? && (fields[5] =~ /^[A-Z][^ ]+$/)
+    metric_key = "metric.#{fields[5].chomp}"
+    metric_key = "#{socket}.#{metric_key}" if socket
+    metric_val = fields[4]
+    metric_val = metric_val.to_f
+  end
+
+  [metric_key, metric_val]
+end
+
+def update_keys(unit, key, socket)
+  socket_key = nil
```

Comment:
thanks, this line is not needed. The later code ```socket_key = "#{socket}.#{key}" if socket``` will set socket_key as nil if socket is nil.